test/e2e/workflow/deployment_test.go

```diff
@@ -281,6 +282,34 @@ var _ = Describe("NetworkAddonsConfig", func() {
 			CreateConfig(gvk, configSpec)
 			CheckConfigCondition(gvk, ConditionAvailable, ConditionTrue, 15*time.Minute, 30*time.Second)
 		})
+		Context("and checking CNAO prometheus endpoint", func() {
+			var err error
+			var scrapedData string
+			BeforeEach(func() {
+				scrapedData, err = GetScrapedDataFromMonitoringEndpoint()
```

Comment:
Why to scrap it on the before each and not in the test body itself ?

Reply:
Because the test is about checking that the metric exists with the expected value on the scrape.
it is not about the scraping itself, and so I dubbed it as something needed to be done BeforeEach test..